keras/layers/wrappers.py

```diff
@@ -167,29 +167,34 @@ def compute_output_shape(self, input_shape):
             return (child_output_shape[0], timesteps) + child_output_shape[1:]
         elif isinstance(child_output_shape, list):
             return [(shape[0], timesteps) + shape[1:] for shape in child_output_shape]
-        raise ValueError('child_output_shape should be a tuple (single output) or a list (multiple outputs)')
+        msg = 'output shape should be a tuple (single output) or a list (multiple outputs)'
+        raise ValueError(msg)
 
     def call(self, inputs, training=None, mask=None):
+        class LPWrapper:
+            # hack equivalent to nonlocal keyword
+            # This design was chosen for Python 2.7 compatibility.
+            uses_learning_phase = False
+
         kwargs = {}
         if has_arg(self.layer.call, 'training'):
             kwargs['training'] = training
-        uses_learning_phase = False
 
         input_shape = K.int_shape(inputs)
         if input_shape[0]:
             # batch size matters, use rnn-based implementation
             def step(x, _):
-                nonlocal uses_learning_phase
                 output = self.layer.call(x, **kwargs)
                 if not isinstance(output, list):
                     output = [output]
                 uses_learning_phase = []
-                for outputi in output:
+                for idx, outputi in enumerate(output):
                     if hasattr(outputi, '_uses_learning_phase'):
                         uses_learning_phase.append(outputi._uses_learning_phase or
-                                                   uses_learning_phase)
+                                                   uses_learning_phase[idx])
```

Comment:
What will happen here if you try to train two independent networks with two separate `model.fit` calls in a single process?